server/index.js

```diff
@@ -46,3 +46,36 @@ app.post('/cart', (req, res) => {
     }
   });
 });
+
+// getting all products information
+app.use('/getAll/:id', (req, res) => {
+  // check if the product/data is already saved in saveProducts obj
+  const id = req.params.id;
+  var data = {};
+  if (saveProducts[id]) {
+    data = saveProducts[id];
+    res.status(200).send(data);
+  } else {
+    api.hrapi(`products/${id}`, (err, data1) => {
+      if (err) {
+        res.status(404).send(err);
+      } else {
+        api.hrapi(`products/${id}/styles`, (err, data2) => {
+          if (err) {
+            res.status(404).send(err);
+          } else {
+            api.hrapi(`reviews/meta/?product_id=${id}`, (err, data3) => {
+              if (err) {
+                res.status(404).send(err);
+              } else {
+                var mergedData = {...data1, ...data2, ...data3};
```

Comment:
creative way to pass the data down to send. sort of looks like callback hell though. Is there a way to use Promises? (Sounds like more work though so this is fine as is)